policy/modules/system/authlogin.te

```diff
@@ -21,6 +21,9 @@ attribute nsswitch_domain;
 type auth_cache_t;
 logging_log_file(auth_cache_t)
 
+type auth_lock_t;
```

Comment:
The implication seems to be the reason for this to exist is because systemd will create the lock file on its own even though it never changes /etc/shadow?

Reply:
That's correct. For some reason systemd needs to be able to write to `.pwd.lock`, else a service using dynamic users will fail to start. I found that once this access is granted, systemd itself never actually accesses `/etc/shadow` or `/etc/passwd`. Not sure why this is the case, but this fixes the problem.

Reply:
Here's the related code: https://github.com/systemd/systemd/blob/fee6441601c979165ebcbb35472036439f8dad5f/src/core/dynamic-user.c#L419

I'd prefer this called `shadow_lock_t` so you can see it is related to `shadow_t`. Then the `auth_manage_shadow()` and `auth_rw_shadow()` can have the rules for doing the lock too, so we don't have to call it everywhere we call those interfaces.

Reply:
OK. That makes sense.